mlflow/protos/service.proto

```diff
@@ -1077,6 +1088,53 @@ message GetMetricHistory {
   }
 }
 
+message MetricWithRunId {
+  // Key identifying this metric.
+  optional string key = 1;
+
+  // Value associated with this metric.
+  optional double value = 2;
+
+  // The timestamp at which this metric was recorded.
+  optional int64 timestamp = 3;
+
+  // Step at which to log the metric.
+  optional int64 step = 4 [default = 0];
+
+  // The ID of the run containing the metric
+  optional string run_id = 5;
+}
+
+message GetMetricHistoryBulkInterval {
+  option (scalapb.message).extends = "com.databricks.rpc.RPC[$this.Response]";
+  option (scalapb.message).extends = "com.databricks.mlflow.api.MlflowTrackingMessage";
+
+  // ID(s) of the run(s) from which to fetch metric values. Must be provided.
+  repeated string run_ids = 1;
+
+  // Name of the metric.
+  optional string metric_key = 2 [(validate_required) = true];
```

Comment:
What happens if some runs are missing the specified metric?

- Run 1: has foo
- Run 2: doesn't have foo

Reply:
Then it won't be found? We're filtering by metric_key